LanguageExt.Core/Queue.cs

```diff
@@ -266,6 +270,10 @@ public static class QueueExtensions
         [Pure]
         public static IEnumerable<T> Distinct<T>(Que<T> queue, Func<T, T, bool> compare) =>
             LanguageExt.List.distinct(queue, compare);
+        
+        [Pure]
+        public static IEnumerable<T> Distinct<T, K>(Que<T> queue, Func<T, K> keySelector, Option<Func<K, K, bool>> compare = default(Option<Func<K, K, bool>>)) =>
```

Comment:
Is this needed?  `Que<T>` is already an `IEnumerable`, and therefore the `Distinct` extension method for `IEnumerable` will do the same job.  If it is needed, then it should be `this Que<T>` as the first argument.